projects/extension/src/background/ConnectionManager.ts

```diff
@@ -252,11 +278,18 @@ export class ConnectionManager
       ? this.#networks.filter((n) => n.tabId === tabId)
       : this.#networks
 
-    const addedChain = await this.#client.addChain({
+    const databaseContent = await this.getObjectFromLocalStorage(
+      (name as string).toLowerCase(),
+    )
```

Comment:
This must only be done for the hardcoded chains: Westend, Polkadot, Kusama, Rococo